src/dune_rules/action_builder.ml

```diff
@@ -42,13 +42,8 @@ let dyn_paths paths = dyn_deps (paths >>| fun (x, paths) -> x, Dep.Set.of_files
 let dyn_paths_unit paths = dyn_deps (paths >>| fun paths -> (), Dep.Set.of_files paths)
 
 let contents p =
-  of_thunk
-    { f =
-        (fun _mode ->
-          let open Memo.O in
-          let+ x = Build_system.read_file p in
-          x, Dep.Map.empty)
-    }
+  let* () = Dep.file p |> Dep.Set.singleton |> Build_system.record_deps in
+  of_memo (Build_system.read_file p)
```

Comment:
@rgrinberg This wasn't happening before and Coq rules were relying on it. Removing the recording of the dep fixes #10149.

Reply:
Huh, indeed! Good catch.